### PR TITLE
Add support for `FragmentInterface` in array conditions

### LINKED ISSUE #145: 🐛 Incorrect insertion of FragmentInterface parameters into WHERE IN queries
### No duplicates 🥲.

- [X] I have searched for a similar issue in our bug tracker and didn't find any solutions.

### Database

MySQL

### What happened?

A bug happened! 

`FragmentInterface` parameters are not functioning properly in `WHERE IN` queries:

```php
    // $guid implements FragmentInterface
    $guid1 = Guid::fromString('0189a22f-1bd0-7005-959a-46860de7d841');
    $guid2 = Guid::fromString('015033e7-a2f0-9ad2-fa9e-e075f4d07a6a');
    $ids = [$guid2]; 
    $select = $this->repository
            ->select()
            ->where('categoryGuid', $guid1);
            ->andWhere('someGuid', 'IN', new Parameter($ids))
            ->fetchData();
```            
Actual behavior:

```sql
WHERE `CategoryGuid` = UUID_TO_BIN('018c11c6-0cae-725e-b885-499e53e8ec5f') 
AND SomeGuid` IN ('018c11c6-5657-7fdc-b2fd-7a6909c1683f') 
```

Expected behavior:

All `FragmentInterface` objects correctly inserted into the query (added `UUID_TO_BIN` also in `IN` parameters)

```sql
WHERE `CategoryGuid` = UUID_TO_BIN('018c11c6-0cae-725e-b885-499e53e8ec5f') 
AND SomeGuid` IN (UUID_TO_BIN('018c11c6-5657-7fdc-b2fd-7a6909c1683f')) 
```

### Version

```shell
database 2.0
PHP 8.2
```

### PR DESCRIPTION
**Pull request by msmakouz**
Closes: #145

### CODE REVIEW

File: src/Driver/Compiler.php [hunk 1]
```diff
@@ -475,15 +475,10 @@ protected function condition(QueryParameters $params, Quoter $q, array $context)
 
         $placeholder = '?';
         if ($value->isArray()) {
-            if ($operator === '=') {
-                $operator = 'IN';
-            } elseif ($operator === '!=') {
-                $operator = 'NOT IN';
-            }
```
Comment (roxblnfk) [inline src/Driver/Compiler.php:482]: Need to add tests with these cases

File: src/Driver/Compiler.php [hunk 2]
```diff
@@ -521,4 +516,24 @@ protected function optional(string $prefix, string $expression, string $postfix
 
         return $prefix . $expression . $postfix;
     }
+
+    private function compileArrayCondition(array $values, QueryParameters $params, Quoter $q, bool $in): string
```
Comment (roxblnfk) [inline src/Driver/Compiler.php]:
```suggestion
    private function arrayToInOperator(QueryParameters $params, Quoter $q, array $values, bool $in): string
```

File: src/Driver/Compiler.php [hunk 3]
```diff
@@ -521,4 +516,24 @@ protected function optional(string $prefix, string $expression, string $postfix
...
+            } else {
+                $placeholders[] = '?';
+                $simpleParams[] = $value;
+            }
+        }
+        if ($simpleParams !== []) {
+            $params->push(new Parameter($simpleParams));
+        }
+
+        return \sprintf('%s (%s)', $operator, \implode(', ', $placeholders));
```
Comment (roxblnfk) [inline src/Driver/Compiler.php]:
```suggestion
        return \sprintf('%s(%s)', $operator, \implode(',', $placeholders));
```
to minimize the result query

File: src/Driver/Compiler.php (hunk 2 above)
Comment (msmakouz) [reply src/Driver/Compiler.php]: Thanks, changed

File: src/Driver/Compiler.php (hunk 3 above)
Comment (msmakouz) [reply src/Driver/Compiler.php]: Thanks, changed

File: src/Driver/Compiler.php (hunk 1 above)
Comment (msmakouz) [reply src/Driver/Compiler.php:482]: Thanks, added.

File: src/Driver/Compiler.php [hunk 4]
```diff
@@ -475,15 +475,10 @@ protected function condition(QueryParameters $params, Quoter $q, array $context)
 
         $placeholder = '?';
         if ($value->isArray()) {
-            if ($operator === '=') {
-                $operator = 'IN';
-            } elseif ($operator === '!=') {
-                $operator = 'NOT IN';
-            }
+            return $this->arrayToInOperator($params, $q, $value->getValue(), $operator === 'IN' || $operator === '=');
```
Comment (roxblnfk) [inline src/Driver/Compiler.php]: It's better to use match-case there and throw an exception if we get an unsupported operator

Comment (roxblnfk) [reply src/Driver/Compiler.php]: I'll do it

**Review by roxblnfk — APPROVED** [review]

**Pull request merged**